assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -5075,4 +5040,482 @@ public String intToTime(int time) {
+                        if (grade != null && gradingService.isUserAbleToGradeItemForStudent(siteId, associatedGradebookAssignmentId, submitterId)) {
+                            gradingService.setAssignmentScoreString(siteId, associatedGradebookAssignmentId, submitterId, grade, "");
+                            String comment = StringUtils.isNotEmpty(cm.get(submitterId)) ? cm.get(submitterId) : "";
+                            gradingService.setAssignmentScoreComment(siteId, associatedGradebookAssignmentId, submitterId, comment);
+                        }
+                    }
+                }
+            } else if (isExternalAssignmentDefined) {
+                gradingService.updateExternalAssessmentScoresString(siteId, ref, sm);
+                gradingService.updateExternalAssessmentComments(siteId, ref, cm);
+            }
+        }
+    }
+
+    private void zeroSubmissionGradingScores(AssignmentSubmission submission, String assignmentRef, String associatedGradebookAssignment, boolean hasExternalAssignment, boolean isGradingItemDefined) {
+
+        if (submission == null) return;
+
+        for (AssignmentSubmissionSubmitter submitter : submission.getSubmitters()) {
+            String submitterId = submitter.getSubmitter();
+            String siteId = submission.getAssignment().getContext();
+            if (hasExternalAssignment) {
+                // external assignment
+                gradingService.updateExternalAssessmentScore(siteId, assignmentRef, submitterId, null);
+            } else if (isGradingItemDefined) {
+                // gb assignment
+                Long associatedGradebookAssignmentId = gradingService.getAssignment(siteId, associatedGradebookAssignment).getId();
+                if (gradingService.isUserAbleToGradeItemForStudent(siteId, associatedGradebookAssignmentId, submitterId)) {
+                    gradingService.setAssignmentScoreString(siteId, associatedGradebookAssignment, submitterId, "0", "");
+                }
+            }
+        }
+    }
+
+    @Override
+    public void zeroAssignmentGradingScores(Assignment assignment) {
+
+        String associatedGradingItem = assignment.getProperties().get(PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT);
+        boolean hasExternalAssignment = gradingService.isExternalAssignmentDefined(assignment.getContext(), associatedGradingItem);
+        String aReference = AssignmentReferenceReckoner.reckoner().assignment(assignment).reckon().getReference();
+        boolean isGradingItemDefined = gradingService.isAssignmentDefined(assignment.getContext(), associatedGradingItem);
+        assignment.getSubmissions().forEach(s -> zeroSubmissionGradingScores(s, aReference, associatedGradingItem, hasExternalAssignment, isGradingItemDefined));
+    }
+
+    @Override
+    @Transactional
+    public void setGradingItemScore(String siteId, String assignmentRef, String associateGradebookAssignment, String submissionRef) {
```

Comment:
I don't think this is a bean pattern so avoid using set and choose something like "saveScore"